File: src/main/java/com/example/finsentinel/service/trading/uta/SecurityType.java
package com.example.finsentinel.service.trading.uta;

public enum SecurityType {
    STOCK,    // Equities (Alpaca, Paper)
    OPTION,   // Equity options (future: IBKR)
    FUTURE,   // Commodity/index futures (future: IBKR)
    CRYPTO,   // Crypto spot (CCXT exchanges)
    PERP,     // Crypto perpetual swap (OKX)
    FOREX;    // Foreign exchange (future: IBKR)

    public boolean isCrypto() {
        return this == CRYPTO || this == PERP;
    }
}

File: packages/shared/src/enums/security-type.ts
export const SecurityType = {
  STOCK: 'STOCK',
  OPTION: 'OPTION',
  FUTURE: 'FUTURE',
  CRYPTO: 'CRYPTO',
  PERP: 'PERP',
  FOREX: 'FOREX',
} as const;

export type SecurityType = (typeof SecurityType)[keyof typeof SecurityType];

/** Returns true if the security type is crypto-related (CRYPTO or PERP). */
export function isCrypto(secType: SecurityType): boolean {
  return secType === SecurityType.CRYPTO || secType === SecurityType.PERP;
}
